Source/ASDisplayNode.mm

```diff
@@ -434,6 +434,18 @@ - (void)onDidLoad:(ASDisplayNodeDidLoadBlock)body
   }
 }
 
+
+- (void)asyncTraitCollectionDidChangeWithPreviousTraitCollection:(ASPrimitiveTraitCollection)previousTraitCollection
+{
+  if (@available(iOS 13.0, *)) {
+    __instanceLock__.lock();
+    if (self.primitiveTraitCollection.userInterfaceStyle != previousTraitCollection.userInterfaceStyle) {
+      __instanceLock__.unlock();
+      [self setNeedsDisplay];
```

Comment:
This will not unlock if they are the same.

Reply:
oh this will unlock when it goes out of scope. the explicit `unlock` in the `if` statement threw me.